lib/src/repo.rs

```diff
@@ -1340,53 +1340,67 @@ pub enum CheckOutCommitError {
 }
 
 mod dirty_cell {
-    use std::cell::{Cell, RefCell};
+    use std::cell::{OnceCell, RefCell};
 
     /// Cell that lazily updates the value after `mark_dirty()`.
+    ///
+    /// A clean value can be immutably borrowed within the `self` lifetime.
     #[derive(Clone, Debug)]
     pub struct DirtyCell<T> {
-        value: RefCell<T>,
-        dirty: Cell<bool>,
+        // Either clean or dirty value is set. The value is boxed to reduce stack space
```

Comment:
The description makes me think it should be an enum with `Clean` and `Dirty` variants. Would that work?

Reply:
No, not without unsafe. The enum would have to be wrapped with `RefCell` or something, which means the inner value can't be borrowed as `(&'a self) -> &'a T`.